manim/mobject/mobject.py

```diff
@@ -2312,14 +2366,90 @@ def split(self) -> list[Self]:
+
+        This method must be called after any change which involves modifying
+        some Mobject's submobjects, such as a call to Mobject.add.
+
+        Parameters
+        ----------
+        only_changed_order
+            If True, indicate that the family still contains the same Mobjects,
+            only in a different order. This prevents recalculating bounding
+            boxes and updater statuses, because they remain the same. If False,
+            indicate that some Mobjects were added or removed to the family,
+            and trigger the aforementioned recalculations. Default is False.
+
+        Returns
+        -------
+        :class:`Mobject`
+            The Mobject itself.
+        """
+        self._family = None
+        # TODO: Implement when bounding boxes and updater statuses are implemented
+        # if not only_changed_order:
+        #     self.refresh_has_updater_status()
+        #     self.refresh_bounding_box()
+        for parent in self.parents:
+            parent.note_changed_family(only_changed_order)
+        return self
+
     def get_family(self, recurse: bool = True) -> list[Self]:
-        sub_families = [x.get_family() for x in self.submobjects]
-        all_mobjects = [self] + list(it.chain(*sub_families))
-        return remove_list_redundancies(all_mobjects)
+        """Obtain the family of this Mobject, consisting of itself, its
+        submobjects, the submobjects' submobjects, and so on. If the
+        family was calculated previously and memoized into the :attr:`family`
+        attribute as a list, return that list. Otherwise, if the attribute is
+        None, calculate and memoize it now.
+
+        Parameters
+        ----------
+        recurse
+            If True, explore this Mobject's submobjects and so on, to
+            compute the full family. Otherwise, stop at this Mobject and
+            return a list containing only this one. Default is True.
+
+        Returns
+        -------
+        list[:class:`Mobject`]
+            The family of this Mobject.
+        """
+        if not recurse:
+            return [self]
+        if self._family is None:
+            # Reconstruct and save
+            sub_families = (sm.get_family() for sm in self.submobjects)
+            family = [self, *it.chain(*sub_families)]
+            self._family = remove_list_redundancies(family)
+        return self._family
 
     def family_members_with_points(self) -> list[Self]:
         return [m for m in self.get_family() if m.get_num_points() > 0]
 
+    def get_ancestors(self, extended: bool = False) -> list[Mobject]:
+        """
+        Returns parents, grandparents, etc.
+        Order of result should be from higher members of the hierarchy down.
+
+        If extended is set to true, it includes the ancestors of all family members,
+        e.g. any other parents of a submobject.
+        """
+        ancestors = []
+        to_process = self.get_family(recurse=extended).copy()
```

Comment:
In this case `extended` has a different meaning.

If it had the same meaning as `get_family()`'s `recurse`, then it would mean something like "if `recurse=False` return a list with only this `Mobject`, otherwise keep recursing through this `Mobject`'s parents, its parents' parents, etc... and return a list containing all of them".

However, this method always returns the ancestors of this `Mobject`.
`extended` here means "if `extended=False` only return the ancestors of this `Mobject`. Otherwise, get the entire family, and then also return the ancestors of all the family members". It's more in the sense of "extended family", in this case meaning your spouse, parents-in-law, children-in-law, etc.

Reply:
In any case, I just copy-pasted this from ManimGL TBH, and it's currently not being used anywhere 🤷

Reply:
Ah that makes sense - maybe it's a good idea to document this somewhere and/or add tests?